addons/account/report/account_invoice_report.py

```diff
@@ -114,8 +114,7 @@ def _sub_select(self):
                     ai.partner_bank_id,
                     SUM ((invoice_type.sign_qty * ail.quantity) / COALESCE(u.factor,1) * COALESCE(u2.factor,1)) AS product_qty,
                     SUM(ail.price_subtotal_signed * invoice_type.sign) AS price_total,
-                    ai.amount_total_signed / (SELECT count(*) FROM account_invoice_line l where invoice_id = ai.id) *
-                    count(*) * invoice_type.sign AS amount_total,
+                    ail.price_total * invoice_type.sign AS amount_total,
```

Comment:
no `SUM`? I would have expected something equivalent to to the line above. Moreover, `price_total` doesn't have the same sign than `price_subtotal_signed`:

https://github.com/odoo/odoo/blob/8f21148e1a8eaae3bc09ec54377becb2361e7834/addons/account/models/account_invoice.py#L1632-L1645

Damn, that seems to be so confusing...

Reply:
For the record: after checking the 4 cases (invoice, refund, vendor bill, credit notes) some signs weren't correct indeed; I instead used sign_qty which was defined by Simon and matches in all cases.
The sum was added because using sign_qty requires to be under an aggregate function.